core/drivers/crypto/se050/core/rng.c

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <crypto/crypto.h>
+#include <kernel/panic.h>
+#include <rng_support.h>
+#include <se050.h>
+#include <tee/tee_cryp_utl.h>
+
+static TEE_Result do_rng_read(void *buf, size_t blen)
+{
+	sss_se05x_rng_context_t rng = { 0 };
+	sss_status_t status = kStatus_SSS_Success;
+
+	sss_se05x_rng_context_init(&rng, se050_session);
+
+	status = sss_se05x_rng_get_random(&rng, buf, blen);
+	if (status != kStatus_SSS_Success)
+		return TEE_ERROR_GENERIC;
```

Comment:
seems `sss_se05x_rng_context_free(&rng)` should be called before returning.

Reply:
ok

Reply:
suggest:
```c
	status = sss_se05x_rng_get_random(&rng, buf, blen);

	sss_se05x_rng_context_free(&rng);

	if (status != kStatus_SSS_Success)
		return TEE_ERROR_GENERIC;

	return TEE_SUCCESS;